packages/nextjs/src/server/utils/withClerkMiddleware.ts

```diff
@@ -0,0 +1,132 @@
+import { AuthStatus } from '@clerk/backend-core';
+import { NextMiddleware, NextMiddlewareResult } from 'next/dist/server/web/types';
+import { NextFetchEvent, NextRequest, NextResponse } from 'next/server';
+
+import { vercelEdgeBase } from '../../edge-middleware';
+import {
+  AUTH_RESULT,
+  AuthResult,
+  NEXT_REDIRECT_HEADER,
+  NEXT_RESUME_HEADER,
+  NEXT_REWRITE_HEADER,
+  SESSION_COOKIE_NAME,
+} from '../types';
+import { getAuthResultFromRequest, setAuthResultOnRequest } from './requestResponseUtils';
+
+const DEFAULT_API_URL = process.env.CLERK_API_URL || 'https://api.clerk.dev';
+const DEFAULT_API_VERSION = process.env.CLERK_API_VERSION || 'v1';
+
+// Using public interstitial endpoint for time being
+const INTERSTITIAL_URL = `${DEFAULT_API_URL}/${DEFAULT_API_VERSION}/public/interstitial?frontendApi=${process.env.NEXT_PUBLIC_CLERK_FRONTEND_API}`;
+
+type WithAuthOptions = {
+  jwtKey?: string;
+  authorizedParties?: string[];
+};
+
+export function withClerkMiddleware(handler: NextMiddleware, opts: WithAuthOptions = {}): NextMiddleware {
+  return async (req: NextRequest, event: NextFetchEvent) => {
+    const { headers, cookies } = req;
+    const { jwtKey, authorizedParties } = opts;
+
+    // Skip if this is an internal next request
+    if (req.nextUrl.pathname.startsWith('/_next/')) {
```

Comment:
💡 It would be great to have an array of path allowlists in `WithAuthOptions` so that if we miss a `Next.js` internal path in the current or future version of the framework, users can still use this library until we patch it.

This is a very common pattern in middleware. We did the same in Ruby last week.

For example:

```ts
type WithAuthOptions = {
  jwtKey?: string;
  authorizedParties?: string[];
  ingoredPaths?: string[];
};

...

if ((req.nextUrl.pathname.startsWith('/_next/') || ingoredPaths.find(p => new Regexp(p).test(req.nextUrl.pathname))) {
  return NextResponse.next()
}
```

Reply:
Adding